src/views/ApartmentApp/components/StaffMenu/index.js

```diff
@@ -57,42 +58,33 @@ const StaffMenu = ({ userProfile, authentication }) => {
     setDateStr(dateStr);
   }, [userProfile, loadAllCurrentApplications]);
 
-  /**
-   * Generate arrays of objects to be converted to a CSV
-   * @param {ApplicationDetails[]} applicationDetailsArray an array of ApplicationDetails objects
-   */
-  const generateCSVData = useCallback((applicationDetailsArray) => {
-    let applicationsForCsv = [];
-    let applicantsForCsv = [];
-    let apartmentChoicesForCsv = [];
-    applicationDetailsArray.forEach(({ Applicants, ApartmentChoices, ...applicationDetails }) => {
-      // Only add the applications that have been submitted
-      if (applicationDetails.DateSubmitted) {
-        applicationsForCsv.push(applicationDetails);
-
-        Applicants.forEach(({ Profile, ...applicant }) =>
-          applicantsForCsv.push({
-            ApplicationID: applicant.ApplicationID ?? applicationDetails.ApplicationID,
-            ...applicant,
-          }),
-        );
-
-        ApartmentChoices.forEach((apartmentChoice) => {
-          apartmentChoicesForCsv.push({
-            ApplicationID: apartmentChoice.ApplicationID ?? applicationDetails.ApplicationID,
-            ...apartmentChoice,
-          });
-        });
-      }
-    });
-    setApplicationJsonArray(applicationsForCsv);
-    setApplicantJsonArray(applicantsForCsv);
-    setApartmentChoiceJsonArray(apartmentChoicesForCsv);
-  }, []);
-
   useEffect(() => {
-    generateCSVData(applications);
-  }, [applications, generateCSVData]);
+    setApplicationJsonArray(
+      applications
+        ?.filter((applicationDetails) => applicationDetails?.DateSubmitted) // Only add the applications that have been submitted
+        ?.map(({ Applicants, ApartmentChoices, ...applicationDetails }) => {
+          const applicationID = applicationDetails.ApplicationID;
```

Comment:
It might be more clear to use `applicationDetails.ApplicationID` directly in the two spots below, rather than storing it as this `applicationID` variable. That way, you can clearly see the distinction between the ID coming from the application versus the ID coming from the ApartmentChoice/Applicant

I'm still wondering though - shouldn't it be impossible to have a mismatch between these? Like, the application ID DEFINES the relationship between these entities, so an Applicant with no application ID should not belong to an application, right?

Reply:
That is true, I think that was because the Applicant object on the frontend didn't originally contain the ApplicationID, which A.) should be something that I add to the StudentApplication code and B.) would not effect the StaffMenu anyway, so I think you are correct that this is no longer needed

I think another reason was that the Applicants array is an array within the ApartmentApplication, so it must always correspond to the same ApplicationID, and I wasn't clear until now that we would include the ApplicationID within the Applicant object received from the API. In their combine/nested form, the ApplicationID in the Applicant objects is redundant because it is nested within the ApartmentApplication object, but the ApplicationID is a necessity when converting it to a CSV